crates/bevy_ui/src/render/mod.rs

```diff
@@ -879,6 +884,70 @@ pub fn extract_text_shadows(
     }
 }
 
+pub fn extract_text_background_colors(
+    mut commands: Commands,
+    mut extracted_uinodes: ResMut<ExtractedUiNodes>,
+    uinode_query: Extract<
+        Query<(
+            Entity,
+            &ComputedNode,
+            &GlobalTransform,
+            &InheritedVisibility,
+            Option<&CalculatedClip>,
+            &ComputedNodeTarget,
+            &TextLayoutInfo,
+        )>,
+    >,
+    text_background_colors_query: Extract<Query<&TextBackgroundColor>>,
+    camera_map: Extract<UiCameraMap>,
+) {
+    let mut camera_mapper = camera_map.get_mapper();
+    for (entity, uinode, global_transform, inherited_visibility, clip, camera, text_layout_info) in
+        &uinode_query
+    {
+        // Skip if not visible or if size is set to zero (e.g. when a parent is set to `Display::None`)
+        if !inherited_visibility.get() || uinode.is_empty() {
+            continue;
+        }
+
+        let Some(extracted_camera_entity) = camera_mapper.map(camera) else {
+            continue;
+        };
+
+        let size = -0.5 * uinode.size().extend(0.);
+        let transform = global_transform.affine() * bevy_math::Affine3A::from_translation(size);
```

Comment:
`size` is a bit unclear for me as it's used as a translation. I would remove the temporary
```suggestion
        let transform = global_transform.affine() * bevy_math::Affine3A::from_translation(-0.5 * uinode.size().extend(0.));
```
or rename it
```suggestion
        let size_offset = -0.5 * uinode.size().extend(0.);
        let transform = global_transform.affine() * bevy_math::Affine3A::from_translation(size_offset);
```
(Even better would be to have a comment or helper to compute this offset, because it seems to be used at least 3 times in the module)